### ISSUE TITLE
Version 0.1.7 breaks integration

### ISSUE THREAD

**Issue by EiNSTeiN-** [gazoodle/gecko-home-assistant#53]
## Version of the custom_component

Version 0.1.7

## Configuration

Not sure what goes here, there's no special configuration. Let me know any details I can add.

## Describe the bug

All entities are unavailable with version 0.1.7 (see screenshot). After downgrading back to 0.1.6 everything works again as expected.
![image](https://user-images.githubusercontent.com/2053039/163582925-a3437549-cc31-4911-8273-eb9dd18814b9.png)

## Debug log

I enabled debug logs and here's everything from gecko:

```text
2022-04-15 10:30:58 INFO (MainThread) [custom_components.gecko]
-------------------------------------------------------------------
Gecko
Version: 0.1.7
Gecko Lib: 0.4.6
This is a custom integration!
If you have any issues with this you need to open an issue here:
https://github.com/gazoodle/gecko-home-assistant/issues
-------------------------------------------------------------------
2022-04-15 10:30:58 DEBUG (MainThread) [custom_components.gecko] Setup entry for UUID 6f1abf74-d4a6-43ed-9263-e7c8eddae101, ID SPA80:1f:12:43:ce:c0, address None (None)
2022-04-15 10:30:58 DEBUG (MainThread) [custom_components.gecko.spa_manager] Event: GeckoSpaEvent.SPA_MAN_ENTER, state GeckoSpaState.IDLE
2022-04-15 10:30:58 DEBUG (MainThread) [custom_components.gecko.spa_manager] Event: GeckoSpaEvent.LOCATING_STARTED, state GeckoSpaState.LOCATING_SPAS
2022-04-15 10:31:00 DEBUG (MainThread) [custom_components.gecko.spa_manager] Event: GeckoSpaEvent.LOCATING_DISCOVERED_SPA, state GeckoSpaState.LOCATING_SPAS
2022-04-15 10:31:03 DEBUG (MainThread) [custom_components.gecko.spa_manager] Event: GeckoSpaEvent.LOCATING_FINISHED, state GeckoSpaState.LOCATED_SPAS
2022-04-15 10:31:03 DEBUG (MainThread) [custom_components.gecko.spa_manager] Event: GeckoSpaEvent.LOCATING_STARTED, state GeckoSpaState.LOCATING_SPAS
2022-04-15 10:31:05 DEBUG (MainThread) [custom_components.gecko.spa_manager] Event: GeckoSpaEvent.LOCATING_DISCOVERED_SPA, state GeckoSpaState.LOCATING_SPAS
2022-04-15 10:31:05 DEBUG (MainThread) [custom_components.gecko.spa_manager] Event: GeckoSpaEvent.LOCATING_FINISHED, state GeckoSpaState.LOCATED_SPAS
2022-04-15 10:31:05 DEBUG (MainThread) [custom_components.gecko.spa_manager] Event: GeckoSpaEvent.CLIENT_HAS_STATUS_SENSOR, state GeckoSpaState.LOCATED_SPAS
2022-04-15 10:31:05 DEBUG (MainThread) [custom_components.gecko.spa_manager] Event: GeckoSpaEvent.CLIENT_HAS_RECONNECT_BUTTON, state GeckoSpaState.CONNECTING
2022-04-15 10:31:05 DEBUG (MainThread) [custom_components.gecko.spa_manager] Event: GeckoSpaEvent.CONNECTION_STARTED, state GeckoSpaState.CONNECTING
2022-04-15 10:31:06 DEBUG (MainThread) [custom_components.gecko.spa_manager] Event: GeckoSpaEvent.RUNNING_PING_RECEIVED, state GeckoSpaState.CONNECTING
2022-04-15 10:31:06 DEBUG (MainThread) [custom_components.gecko.spa_manager] Event: GeckoSpaEvent.CONNECTION_GOT_FIRMWARE_VERSION, state GeckoSpaState.CONNECTING
2022-04-15 10:31:07 DEBUG (MainThread) [custom_components.gecko.spa_manager] Event: GeckoSpaEvent.CLIENT_HAS_PING_SENSOR, state GeckoSpaState.CONNECTING
2022-04-15 10:31:07 DEBUG (MainThread) [custom_components.gecko.spa_manager] Event: GeckoSpaEvent.CONNECTION_GOT_CHANNEL, state GeckoSpaState.CONNECTING
2022-04-15 10:31:08 DEBUG (MainThread) [custom_components.gecko.spa_manager] Event: GeckoSpaEvent.CONNECTION_GOT_CONFIG_FILES, state GeckoSpaState.CONNECTING
2022-04-15 10:31:08 DEBUG (MainThread) [custom_components.gecko.spa_manager] Event: GeckoSpaEvent.CONNECTION_INITIAL_DATA_BLOCK_REQUEST, state GeckoSpaState.CONNECTING
2022-04-15 10:31:12 DEBUG (MainThread) [custom_components.gecko.spa_manager] Event: GeckoSpaEvent.CONNECTION_SPA_COMPLETE, state GeckoSpaState.SPA_READY
2022-04-15 10:31:12 DEBUG (MainThread) [custom_components.gecko.spa_manager] Event: GeckoSpaEvent.CLIENT_FACADE_IS_READY, state GeckoSpaState.CONNECTED
2022-04-15 10:31:12 DEBUG (MainThread) [custom_components.gecko.spa_manager] Event: GeckoSpaEvent.CONNECTION_FINISHED, state GeckoSpaState.CONNECTED
2022-04-15 10:32:06 DEBUG (MainThread) [custom_components.gecko.spa_manager] Event: GeckoSpaEvent.RUNNING_PING_RECEIVED, state GeckoSpaState.CONNECTED
```

**Comment by EiNSTeiN-** [comment]
This might not be an issue with 0.1.7, although I never had the issue before upgrading to this version. Here's what I noticed:
1. upgraded to 0.1.7, noticed entities are unavailable
2. downgraded to 0.1.6 and noticed everything was back to normal
3. upgraded again to get logs and open the issue
4. downgraded to 0.1.6 again, and now entities are still unavailable. I'm unsure why, I restarted the in.touch and home assistant, nothing works.

**Comment by EiNSTeiN-** [comment]
On 0.1.7 I'm getting a lot of these messages which somehow didn't appear when I got the logs to open this issue, hope this is useful:

```text
2022-04-15 11:03:39 WARNING (MainThread) [geckolib.driver.protocol.unhandled] No handler for b"STATV\x06\x07'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x0b\x00\x00\x00\x08\x02\x00\x02\x00\x1e\x00\x00\x00\x00\xff\xff\xff" from ('192.168.1.170', 10022, b'SPA80:1f:12:43:ce:c0', b'IOS6f1abf74-d4a6-43ed-9263-e7c8eddae101') found, message ignored
2022-04-15 11:03:40 WARNING (MainThread) [geckolib.driver.protocol.unhandled] No handler for b"STATV\x07\x08'\x03\x02\x02\xac\x02\xac\x00\x00\x00\x00\x00\x00\x00\x15\x08\x00\x01C\x00**\x00=<\x01L\x02\x00\x00 \x00\x00\x00\x00\x00\x00\x00\x00\x07" from ('192.168.1.170', 10022, b'SPA80:1f:12:43:ce:c0', b'IOS6f1abf74-d4a6-43ed-9263-e7c8eddae101') found, message ignored
2022-04-15 11:03:40 ERROR (MainThread) [custom_components.solark_modbus.hub] Reading realtime data failed! Inverter is unreachable.
2022-04-15 11:03:40 WARNING (MainThread) [geckolib.driver.protocol.unhandled] No handler for b"STATV\x08\t'\x08\x00\x00\x00\x00\x02\xb4\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00" from ('192.168.1.170', 10022, b'SPA80:1f:12:43:ce:c0', b'IOS6f1abf74-d4a6-43ed-9263-e7c8eddae101') found, message ignored
2022-04-15 11:03:41 WARNING (MainThread) [geckolib.driver.protocol.unhandled] No handler for b"STATV\t\n'\x00\x00\x00\x00\x01\xa5\x00\x00\x00\x00\x00\x00\x00\x00u\xba\x00\x00\x00\x00\x12\xc7\x17\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00" from ('192.168.1.170', 10022, b'SPA80:1f:12:43:ce:c0', b'IOS6f1abf74-d4a6-43ed-9263-e7c8eddae101') found, message ignored
2022-04-15 11:03:44 WARNING (MainThread) [geckolib.driver.protocol.unhandled] No handler for b"STATV\n\x0b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00" from ('192.168.1.170', 10022, b'SPA80:1f:12:43:ce:c0', b'IOS6f1abf74-d4a6-43ed-9263-e7c8eddae101') found, message ignored
2022-04-15 11:03:45 WARNING (MainThread) [geckolib.driver.protocol.unhandled] No handler for b"STATV\x0b\x0c'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x01\x02\x03\x04\x05\x06\x07\x08\t\n\x0b\x0c\r\x0e3\x974\x985\x99" from ('192.168.1.170', 10022, b'SPA80:1f:12:43:ce:c0', b'IOS6f1abf74-d4a6-43ed-9263-e7c8eddae101') found, message ignored
2022-04-15 11:03:47 WARNING (MainThread) [geckolib.driver.protocol.unhandled] No handler for b"STATV\x0c\r'6\x9a7\x9b8\x9c9\x9d:\x9e;\x9finXE_C61.xml\x00\x00\x00\x00inXE_S60.xm" from ('192.168.1.170', 10022, b'SPA80:1f:12:43:ce:c0', b'IOS6f1abf74-d4a6-43ed-9263-e7c8eddae101') found, message ignored
2022-04-15 11:03:47 WARNING (MainThread) [homeassistant.components.automation.garage_door_presence] Garage - Door presence: Already running
2022-04-15 11:03:47 WARNING (MainThread) [geckolib.driver.protocol.unhandled] No handler for b"STATV\r\x0e'l\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00" from ('192.168.1.170', 10022, b'SPA80:1f:12:43:ce:c0', b'IOS6f1abf74-d4a6-43ed-9263-e7c8eddae101') found, message ignored
2022-04-15 11:03:48 WARNING (MainThread) [homeassistant.components.automation.master_bedroom_change_alarm_state_with_sleep_mode] Master bedroom - Change sleep mode with alarm state: Already running
2022-04-15 11:03:48 WARNING (MainThread) [homeassistant.components.automation.garage_door_presence] Garage - Door presence: Already running
2022-04-15 11:03:48 WARNING (MainThread) [homeassistant.components.automation.garage_door_presence] Garage - Door presence: Already running
2022-04-15 11:03:48 WARNING (MainThread) [geckolib.driver.protocol.unhandled] No handler for b"STATV\x0e\x0f'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00" from ('192.168.1.170', 10022, b'SPA80:1f:12:43:ce:c0', b'IOS6f1abf74-d4a6-43ed-9263-e7c8eddae101') found, message ignored
2022-04-15 11:03:48 WARNING (MainThread) [geckolib.driver.protocol.unhandled] No handler for b"STATV\x0f\x10'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00" from ('192.168.1.170', 10022, b'SPA80:1f:12:43:ce:c0', b'IOS6f1abf74-d4a6-43ed-9263-e7c8eddae101') found, message ignored
2022-04-15 11:03:48 WARNING (MainThread) [geckolib.driver.protocol.unhandled] No handler for b"STATV\x10\x11'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00" from ('192.168.1.170', 10022, b'SPA80:1f:12:43:ce:c0', b'IOS6f1abf74-d4a6-43ed-9263-e7c8eddae101') found, message ignored
2022-04-15 11:03:48 WARNING (MainThread) [geckolib.driver.protocol.unhandled] No handler for b"STATV\x11\x12'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00" from ('192.168.1.170', 10022, b'SPA80:1f:12:43:ce:c0', b'IOS6f1abf74-d4a6-43ed-9263-e7c8eddae101') found, message ignored
2022-04-15 11:03:49 WARNING (MainThread) [geckolib.driver.protocol.unhandled] No handler for b"STATV\x12\x13'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00" from ('192.168.1.170', 10022, b'SPA80:1f:12:43:ce:c0', b'IOS6f1abf74-d4a6-43ed-9263-e7c8eddae101') found, message ignored
2022-04-15 11:03:49 WARNING (MainThread) [geckolib.driver.protocol.unhandled] No handler for b"STATV\x13\x14'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00" from ('192.168.1.170', 10022, b'SPA80:1f:12:43:ce:c0', b'IOS6f1abf74-d4a6-43ed-9263-e7c8eddae101') found, message ignored
2022-04-15 11:03:49 WARNING (MainThread) [geckolib.driver.protocol.unhandled] No handler for b"STATV\x14\x15'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00" from ('192.168.1.170', 10022, b'SPA80:1f:12:43:ce:c0', b'IOS6f1abf74-d4a6-43ed-9263-e7c8eddae101') found, message ignored
2022-04-15 11:03:49 WARNING (MainThread) [geckolib.driver.protocol.unhandled] No handler for b"STATV\x15\x16'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00" from ('192.168.1.170', 10022, b'SPA80:1f:12:43:ce:c0', b'IOS6f1abf74-d4a6-43ed-9263-e7c8eddae101') found, message ignored
2022-04-15 11:03:49 WARNING (MainThread) [homeassistant.components.mqtt.discovery] Received message on illegal discovery topic 'homeassistant/binary_sensor/flood_machine_café/config'
2022-04-15 11:03:49 WARNING (MainThread) [geckolib.driver.protocol.unhandled] No handler for b"STATV\x16\x17'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00" from ('192.168.1.170', 10022, b'SPA80:1f:12:43:ce:c0', b'IOS6f1abf74-d4a6-43ed-9263-e7c8eddae101') found, message ignored
2022-04-15 11:03:50 WARNING (MainThread) [geckolib.driver.protocol.unhandled] No handler for b"STATV\x17\x18'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00" from ('192.168.1.170', 10022, b'SPA80:1f:12:43:ce:c0', b'IOS6f1abf74-d4a6-43ed-9263-e7c8eddae101') found, message ignored
2022-04-15 11:03:50 WARNING (MainThread) [geckolib.driver.protocol.unhandled] No handler for b"STATV\x18\x19'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00" from ('192.168.1.170', 10022, b'SPA80:1f:12:43:ce:c0', b'IOS6f1abf74-d4a6-43ed-9263-e7c8eddae101') found, message ignored
2022-04-15 11:03:50 WARNING (MainThread) [geckolib.driver.protocol.unhandled] No handler for b"STATV\x19\x1a'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00" from ('192.168.1.170', 10022, b'SPA80:1f:12:43:ce:c0', b'IOS6f1abf74-d4a6-43ed-9263-e7c8eddae101') found, message ignored
2022-04-15 11:03:50 WARNING (MainThread) [geckolib.driver.protocol.unhandled] No handler for b'STATV\x1a\x00\n\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00' from ('192.168.1.170', 10022, b'SPA80:1f:12:43:ce:c0', b'IOS6f1abf74-d4a6-43ed-9263-e7c8eddae101') found, message ignored
2022-04-15 11:03:50 WARNING (MainThread) [geckolib.driver.protocol.unhandled] No handler for b'WCGET\x05' from ('192.168.1.170', 10022, b'SPA80:1f:12:43:ce:c0', b'IOS6f1abf74-d4a6-43ed-9263-e7c8eddae101') found, message ignored
2022-04-15 11:03:50 WARNING (MainThread) [geckolib.driver.protocol.unhandled] No handler for b'WCGET\x05' from ('192.168.1.170', 10022, b'SPA80:1f:12:43:ce:c0', b'IOS6f1abf74-d4a6-43ed-9263-e7c8eddae101') found, message ignored
```

**Comment by EiNSTeiN-** [comment]
This showed up in logs after a while, could also be relevant

```
2022-04-15 11:05:50 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
File "/usr/local/lib/python3.9/site-packages/geckolib/automation/async_facade.py", line 90, in _facade_update
self._water_care.change_watercare_mode(
File "/usr/local/lib/python3.9/site-packages/geckolib/automation/watercare.py", line 83, in change_watercare_mode
self._on_change(self, old_mode, self.active_mode)
File "/usr/local/lib/python3.9/site-packages/geckolib/driver/observable.py", line 39, in _on_change
f"{self.__class__.__name__} {sender} changed "
File "/usr/local/lib/python3.9/site-packages/geckolib/automation/watercare.py", line 90, in __str__
return f"{self.name}: {GeckoConstants.WATERCARE_MODE_STRING[self.active_mode]}"
IndexError: list index out of range
```

**Comment by EiNSTeiN-** [comment]
Small update... I think the culprit was the last exception I posted (`IndexError: list index out of range`). I was running a leak test with the pump turned on 24 hours/day and therefore the spa was in a mode that doesn't map to any of the `WATERCARE_MODE_STRING`.

**Comment by gazoodle** [comment]
I merged your PR, so this should be resolved in v0.1.8 now
